src/invision/components/movie-list/movie-list.js

```diff
@@ -1,90 +1,133 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
 import thumbnail from './thumbnail.png';
 import MovieItem from '../movie-item/movie-item'
 import './movie-list.scss';
 import MovieService from '../../services/movie.service';
 
-export default class MovieList extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      movieList: [],
-      sortValue: 'RELEASE DATE',
-    };
+import MoviePopup from '../movie-popup/movie.popup';
+import DeletePopup from '../delete-movie/delete-popup';
+import useFetchMovie from './fetch-movie.hook';
 
-    this.getMovieList();
+export default function MovieList() {
 
-    this.sort = this.sort.bind(this);
-  }
+  const [movieState, updateMovieState] = useState({
```

Comment:
When you use **useState** hook, you add property by property to the state and specify change function for each property separately:
```  const [movieList, setMovieList] = useState([]);
  const [sortValue, setSortValue] = useState('RELEASE DATE');
  const [isEditMovieVisible, setIsEditMovieVisible] = useState(false);
  const [isDeleteMovieVisible, setIsDeleteMovieVisible] = useState(false);
  const [selectedMovie, setSelectedMovie] = useState(null);
```